src/Storages/IStorageCluster.cpp

```diff
@@ -183,29 +186,43 @@ void ReadFromCluster::initializePipeline(QueryPipelineBuilder & pipeline, const
     auto new_context = updateSettings(context->getSettingsRef());
     const auto & current_settings = new_context->getSettingsRef();
     auto timeouts = ConnectionTimeouts::getTCPTimeoutsWithFailover(current_settings);
+
+    auto max_replicas_to_use = cluster->getShardsInfo().size();
```

Comment:
This looks wrong. `max_parallel_replicas` controls the number of replicas to use in a shard, not among shards.

Reply:
Asked in the corresponding issue - https://github.com/ClickHouse/ClickHouse/issues/65024#issuecomment-2551887448